fedimint-api/src/module/registry.rs

```diff
@@ -0,0 +1,62 @@
+use std::collections::BTreeMap;
+
+use crate::core::Decoder;
+use crate::server::ServerModule;
+
+// TODO: unify and/or make a newtype?
+/// Fedimint module identifier
+pub type ModuleKey = u16;
+
+/// Collection of server modules
+#[derive(Debug, Default, Clone)]
+pub struct ModuleRegistry(BTreeMap<ModuleKey, ServerModule>);
```

Comment:
What's the plan for client side module registry? The `M` was there because of the assumption we might need `ModuleRegistry<ServerModule>` and `ModuleRegistry<ClientModule>`. It might be kind of weird for the client code to initialize server-modules. They might have additional requirements, pull in extra crypto libs etc.

Reply:
I thought about making `ModuleRegistry` generic and then having specializations (server, client, decoder) that are type defs with their own specific impls, might rework it tomorrow, just didn't have time yet. Getting rid of `M` in places where only a decoder is required is the right move though imo (and most of the work in this PR).

Reply:
Maybe we could have `pub struct ModuleRegistry(BTreeMap<ModuleKey, ModuleDecoder>);, and then on both `ServerModule` and `ClientModule` ability to get an instance of `ModuleDecoder`. An extra step, but would unify the two. Possibly the client could even have `unimplemented!();` for things that client is not supposed to decode, if that would require bringing extra dependencies etc.

Reply:
I'll worry about the client when I get to it. That will require some refactoring of common functions between client/server modules. I prepared a stub in this PR now, but I don't want to introduce a bunch of dead code on the client side rn. I'll impl when it's actually needed.